pkg/block/transient/adapter.go

```diff
@@ -25,11 +29,8 @@ func (a *Adapter) Put(_ context.Context, _ block.ObjectPointer, _ int64, reader
 	return err
 }
 
-func (a *Adapter) Get(_ context.Context, _ block.ObjectPointer, expectedSize int64) (io.ReadCloser, error) {
-	if expectedSize < 0 {
-		return nil, io.ErrUnexpectedEOF
-	}
-	return io.NopCloser(&io.LimitedReader{R: rand.Reader, N: expectedSize}), nil
+func (a *Adapter) Get(_ context.Context, _ block.ObjectPointer) (io.ReadCloser, error) {
+	return io.NopCloser(&io.LimitedReader{R: rand.Reader, N: DefaultReaderSize}), nil
```

Comment:
I'm not sure where this type of adapter is used but this changes the behavior in the following way: 

**previous:** 
- if size < 0 return error `io.ErrUnexpectedEOF` an actually error 
- else size always 0 and that means when using the LimitedReader and the error returned is `io.EOF` usually not an error more an indicator.

**now:** 
it will always read data from the rand.Reader  `DefaultReaderSize = 1024 * 1024`  and only then return `io.EOF`. 

I'm fine with this change here, but I suggest to keep it simpler, read anyway but hardcoded `0` and remove `DefaultReaderSize`. WDYT?

Reply:
I don't think it matters because I doubt this adapter is used anywhere. I vote to keep it this way and treat it as a "scream test" :)